### PR TITLE
Escape regex input

### PR DESCRIPTION
**Pull request by sarayourfriend**
Also delete some unused methods and convert the "$regex" objects into proper Patterns (Thanks @Wizard4096 for the tip!)

Also adds a unit test that I think demonstrates a better approach to testing the `api` module than the one we've been taking. Instead of testing private methods we should be able to just test the public surface of the `api` module.

### CODE REVIEW

**Comment by Wizard4096** [comment]
This looks good to me! 👍

**Review by Wizard4096 — APPROVED** [review]

File: tests/test_api.py [hunk 1]
```diff
@@ -1,7 +1,83 @@
...
+from unittest.mock import MagicMock
+
 import api
 from enums import SearchBehavior
 
 
+@pytest.mark.asyncio
+@patch("api._get_entities")
+async def test_search_posts_with_mentions(get_entities):
+    get_entities.return_value = 0, []
```
Comment (AetherUnbound) [inline tests/test_api.py:13]: This doesn't need to be wrapped in parenthesis? :O

File: tests/test_api.py [hunk 2]
```diff
@@ -1,7 +1,83 @@
...
+    mongo = MagicMock()
```
Comment (AetherUnbound) [inline tests/test_api.py:14]: I think we can use `spec=quart_motor.Motor` here for some extra guardrails :) 
https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock

**Review by AetherUnbound — APPROVED** [review]
Nice!

File: tests/test_api.py (hunk 2 above)
Comment (sarayourfriend) [reply tests/test_api.py:14]: Oh cool! I didn't know about that

**Review by AetherUnbound — APPROVED** [review]

**Pull request merged**